### ISSUE TITLE
relay test failure with with hdf5@1.10.7?

### ISSUE THREAD

**Issue by cyrush** [LLNL/conduit#817]
doing a few test builds and a few unit tests failed when building against 1.10.7

investigate.

**Comment by cyrush** [comment]
```
The following tests FAILED:
	 71 - t_relay_io_handle (Failed)
	 76 - t_relay_io_hdf5 (Failed)
	 86 - t_python_relay_io_handle (Failed)
```

t_relay_io_hdf5 failures:
 
```
[  FAILED  ] conduit_relay_io_hdf5.write_and_read_conduit_leaf_to_fixed_hdf5_dataset_handle_with_offset
[  FAILED  ] conduit_relay_io_hdf5.write_conduit_object_to_hdf5_group_handle_with_offset
[  FAILED  ] conduit_relay_io_hdf5.conduit_hdf5_list_with_offset
```

These cases involve `H5Oget_info` not behaving as I would expect. 

fail to fetch info for existing object:

```
//---------------------------------------------------------------------------//
void
read_hdf5_tree_into_conduit_node(hid_t hdf5_id,
                                 const std::string &ref_path,
                                 bool only_get_metadata,
                                 const Node &opts,
                                 Node &dest)
{
    herr_t     h5_status = 0;
    H5O_info_t h5_info_buf;

    h5_status = H5Oget_info(hdf5_id,&h5_info_buf);

    CONDUIT_CHECK_HDF5_ERROR_WITH_FILE_AND_REF_PATH(h5_status,
                                                    hdf5_id,
                                                    ref_path,
                                           "Error fetching HDF5 object "
                                           << "info from: "
                                           << hdf5_id);
```

also fail to fetch info for an existing object (but causes a new object to be created and subsequent confusion)

```
//---------------------------------------------------------------------------//
void
write_conduit_leaf_to_hdf5_group(const Node &node,
                                 const std::string &ref_path,
                                 hid_t hdf5_group_id,
                                 const std::string &hdf5_dset_name,
                                 const Node &opts)
{
    // data set case ...


    // check if the dataset exists
    H5O_info_t h5_obj_info;
    
    herr_t h5_info_status =  H5Oget_info_by_name(hdf5_group_id,
                                                  hdf5_dset_name.c_str(),
                                                  &h5_obj_info,
                                                  H5P_DEFAULT);
```

`h5_info_status` is fail,  even though the object exists (verified by looking at hdf5 file)

**Comment by cyrush** [comment]
the issue is related to magical conversion to an extendable dataset. 

Due to our current logic calling chain,  we can have hdf5 ids (handles) that point to the old data set after it has been closed and a new data set was created.

**Comment by cyrush** [comment]
resolved in #884
